src/SignalR/clients/csharp/Http.Connections.Client/src/HttpConnection.cs

```diff
@@ -461,9 +462,10 @@ private async Task<NegotiationResponse> NegotiateAsync(Uri url, HttpClient httpC
 
             using (var request = new HttpRequestMessage(HttpMethod.Post, uri))
             {
-                // Corefx changed the default version and High Sierra curlhandler tries to upgrade request
-                request.Version = new Version(1, 1);
-
+#if NETSTANDARD2_1_OR_GREATER || NET7_0_OR_GREATER
+                // HttpClient gracefully falls back to HTTP/1.1, so it's fine to set the preferred version to a higher version
+                request.Version = HttpVersion.Version20;
```

Comment:
What happens if we didn't set Version at all and left it default?

Reply:
It defaults to 1.1, and a big benefit of HTTP/2 WS is sharing an existing connection, which it wouldn't be if we used HTTP/1.1
https://source.dot.net/#System.Net.Http/System/Net/Http/HttpRequestMessage.cs,14

Reply:
So we're going to start using HTTP/2 by default for negotiate here and sends by default in SendUtils, but WebSockets and long polls are going to continue to default to HTTP/1.1?

Reply:
Oh, good point about LongPolling, we can switch that to HTTP/2 by default.

But otherwise, what you said is correct.

Reply:
And the SSE event-source request?

Reply:
Doh, yes...